File: src/Services/BookingService/models/booking.ts
import { Column, Entity, JoinColumn, OneToOne, PrimaryGeneratedColumn } from "typeorm";
import { ParkingSlot } from "./parking-slot";


@Entity()
export class Booking {
    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    userId: number;

    @OneToOne(() => ParkingSlot, (slot) => slot.id)
    @JoinColumn()
    parkingSlotId: number;

    @Column()
    startDate: Date;

    @Column()
    endDate: Date;
}

File: src/Services/BookingService.Aspnet/Models/Booking.cs
namespace BookingService.Aspnet.Models;

internal class Booking
{
    public Booking(string userId, ParkingSlot parkingSlot, DateTime startDate, DateTime endDate)
    {
        UserId = userId;
        ParkingSlot = parkingSlot;
        StartDate = startDate;
        EndDate = endDate;
    }

    // For EF
    private Booking()
    {
    }

    public int Id { get; set; }
    public string UserId { get; set; }
    public ParkingSlot ParkingSlot { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
}